### PR TITLE
fix: stop preventing canvas pointerdown/tapend events

### PR DESCRIPTION
**Pull request by dwelle**
Attempt to fix https://github.com/excalidraw/excalidraw/issues/3151

~~Also, this fixes a case where clicking on canvas doesn't deselect selection outside Excalidraw component.~~

### CODE REVIEW

**Comment by dwelle** [comment]
This actually doesn't work well, as it select text outside the component on canvas double-click. Will investigate.

**Comment by dwelle** [comment]
So, I kept not preventing default on `pointerdown`, but had to revert not preventing selection on `canvas` which was also allowing selecting stuff outside the component.

Unfortunately, this means that clicking on canvas still retains selection that's outside the component. If we need that behavior, we'll need to manually deselect on `pointerdown`.

File: src/css/styles.scss [hunk 1]
```diff
@@ -17,6 +17,13 @@
   left: 0;
   right: 0;
 
+  // serves 2 purposes:
+  // 1. prevent selecting text outside the component when double-clicking or
+  //    dragging inside it (e.g. on canvas)
+  // 2. prevent selecting UI, both from the inside, and from outside the
+  //    component (e.g. if you select text in a sidebar)
+  user-select: none;
```
Comment (ad1992) [inline src/css/styles.scss:25]: So with this, the selection will also be prevented in the npm package which was not till now 👍

File: src/components/App.tsx [hunk 2]
```diff
@@ -2109,15 +2109,6 @@ class App extends React.Component<ExcalidrawProps, AppState> {
...
 
-    // fixes pointermove causing selection of UI texts #32
-    event.preventDefault();
-    // Preventing the event above disables default behavior
-    // of defocusing potentially focused element, which is what we
-    // want when clicking inside the canvas.
-    if (document.activeElement instanceof HTMLElement) {
-      document.activeElement.blur();
-    }
-
```
Comment (ad1992) [inline src/components/App.tsx:2120]: Just need to make sure nothing breaks due to this removal else pr looks good

**Review by ad1992 — APPROVED** [review]

File: src/components/App.tsx (hunk 2 above)
Comment (dwelle) [reply src/components/App.tsx:2120]: Definitely do check. I wouldn't be surprised if something did :).

**Review by lipis — APPROVED** [review]

**Comment by ad1992** [comment]
I am facing some issues when trying type text, so pointer doesn't change to cursor with single click on canvas, i have to double click.
Excalidraw (1)
 This works fine in prod.

**Comment by dwelle** [comment]
> I am facing some issues when trying type text, so pointer doesn't change to cursor with single click on canvas, i have to double click.

I "fixed" it by reintroducing the `preventDefault` in case of `text`, and manually blurring in any case. Both of these are pretty intertwined with the logic, some of which I'm not sure even how it works.

If we want to keep this hacks it'll mean that the case this PR was supposed to fix (by *not* preventing the default) will not be 100% fixed (due to the text handling) — well, I think it wouldn't be either way, coz we also prevent default on mobile (touch events) for various reasons.

It could in theory be fixed by delaying (into next event loop tick) binding the wysiwyg blur event.

**Comment by dwelle** [comment]
Breaks mobile. Probably related to the other `preventDefault` calls we do for touch events.

**Comment by dwelle** [comment]
Removing the `preventDefault` in `onTapEnd` handler seems to have fixed the mobile issue (which was preventing wysiwyg submit when clicking outside).

**Review by ad1992 — APPROVED** [review]

File: src/css/styles.scss (hunk 1 above)
Comment (craigkovatch) [reply src/css/styles.scss:25]: @dwelle does the SCSS preprocessor handle vendor prefixing of this? If not, still need `-webkit-user-select` fallback for several browsers

**Pull request merged**